core/src/dealer/maps.rs

```diff
@@ -0,0 +1,116 @@
+use std::collections::HashMap;
+
+pub struct AlreadyHandledError(());
+
+pub enum HandlerMap<T> {
+    Leaf(T),
+    Branch(HashMap<String, HandlerMap<T>>),
+}
```

Comment:
We could use [`DashMap`](https://github.com/xacrimon/dashmap) to avoid a mutex around this struct. There would be a lot of places where librespot could benefit of `dashmap`, so I leave it for a future PR.